src/useSearchParamInput.ts

```diff
@@ -0,0 +1,59 @@
+import { useState, useEffect } from 'react';
+
+/**
+ * Taken from ReactTraining/history: https://github.com/ReactTraining/history/blob/master/modules/DOMUtils.js
+ *
+ * Returns true if the HTML5 history API is supported. Taken from Modernizr.
+ *
+ * https://github.com/Modernizr/Modernizr/blob/master/LICENSE
+ * https://github.com/Modernizr/Modernizr/blob/master/feature-detects/history.js
+ * changed to avoid false negatives for Windows Phones: https://github.com/reactjs/react-router/issues/586
+ */
+function supportsHistory() {
+  const ua = window.navigator.userAgent;
+
+  if (
+    (ua.indexOf('Android 2.') !== -1 || ua.indexOf('Android 4.0') !== -1) &&
+    ua.indexOf('Mobile Safari') !== -1 &&
+    ua.indexOf('Chrome') === -1 &&
+    ua.indexOf('Windows Phone') === -1
+  ) {
+    return false;
+  }
+
+  return window.history && 'pushState' in window.history;
+}
+
+function useSearchParamInput(
+  key: string,
+  defaultValue?: string,
```

Comment:
Shall we make this type `null | string`, then if `typeof defaultValue !== 'string'` use the global value, otherwise use and set the provided `defaultValue` string.

Actually, do we need the `defaultValue`? Maybe user could themselves set it?

```ts
const [value, setValue] = useSearchParamInput('gg');
useEffect(() => {
  if (value === null) ...
}, []);
```

Reply:
> Shall we make this type `null | string`, then if `typeof defaultValue !== 'string'`

Sure!

> Actually, do we need the defaultValue? Maybe user could themselves set it?

I think having a `defaultValue` would follow the same React usage of `useState`